File: ios/SwiftKotlination/SwiftKotlination/Sources/Features/Story/Model/StoryRepository.swift
protocol StoryRepositoryProtocol {
    func story(_ observer: @escaping Observer<Story>)
}

struct StoryRepository {
    let story: Story
}

// MARK: - Protocol Methods

extension StoryRepository: StoryRepositoryProtocol {
    func story(_ observer: @escaping Observer<Story>) {
        observer(.success(story))
    }
}

File: android/SwiftKotlination/app/src/main/java/fr/jhandguy/swiftkotlination/features/story/model/StoryRepository.kt
package fr.jhandguy.swiftkotlination.features.story.model

import fr.jhandguy.swiftkotlination.Result

interface StoryRepository {
    suspend fun story(observer: (Result<Story>) -> Unit)
}

class StoryRepositoryImpl(private val story: Story): StoryRepository {
    override suspend fun story(observer: (Result<Story>) -> Unit) = observer(Result.Success(story))
}